core/container.go

```diff
@@ -560,6 +561,8 @@ func (container *Container) WithMountedFile(ctx context.Context, bk *buildkit.Cl
 	return container.withMounted(ctx, bk, target, file.LLB, file.File, file.Services, owner)
 }
 
+var SeenCacheKeys sync.Map
```

Comment:
nit: I think we should be paranoid here and make this a pointer, just in case the map gets passed to a goroutine or func somewhere, since we don't want it to ever be copied.

```suggestion
var SeenCacheKeys = new(sync.Map)
```